### PR TITLE
feat: eip-712 strucured signing

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Web companion PR of https://github.com/shapeshift/hdwallet/pull/673

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/5916

Low - the only affected path is CoW signing, which should be tested with all wallets, and look much better now!

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

See https://github.com/shapeshift/hdwallet/pull/673

### CODE REVIEW

File: packages/swapper/src/types.ts [hunk 1]
```diff
@@ -345,6 +347,7 @@ export type EvmTransactionRequest = {
 
 export type CowMessageToSign = {
   chainId: ChainId
+  from: string
```
Comment (kaladinlight) [inline packages/swapper/src/types.ts:350]: not seeing this used anywhere

Comment (gomesalexandre) [reply packages/swapper/src/types.ts]: nice catch! Legacy from first fiddlings with this - 167b858012

**Review by kaladinlight — APPROVED** [review]
Tested native and MM, looks gucci!

**Comment by gomesalexandre** [comment]
Moved to draft again so this doesn't get merged - trying to keep this release portals-y

**Pull request merged**